### PR TITLE
Merged datasource improvements

### PR DESCRIPTION
**Pull request by alesgenova**
Improvements to the DataPropertiesPanel:
- Selection of active scalars from table and combo box
- Renaming of scalars array in the active datasource

screenshot from 2018-11-15 15-11-52

Fixes: #1741 
Fixes: #1738 
Fixes: #1739

### CODE REVIEW

File: tomviz/DataPropertiesModel.cxx [hunk 1]
```diff
@@ -0,0 +1,116 @@
+/* This source file is part of the Tomviz project, https://tomviz.org/.
+   It is released under the 3-Clause BSD License, see "LICENSE". */
+
+#include "DataPropertiesModel.h"
+
+#include <QBrush>
```
Comment (cjh1) [inline tomviz/DataPropertiesModel.cxx]: I don't think this is used?

File: tomviz/DataPropertiesModel.cxx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+#include <QDebug>
```
Comment (cjh1) [inline tomviz/DataPropertiesModel.cxx]: Likewise?

File: tomviz/DataPropertiesModel.h [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  void activeScalarsChanged(QString name);
+  void scalarsRenamed(QString oldNama, QString newName);
+
+private:
+  QList<ArrayInfo> m_arraysInfo;
+  const int c_numCol = 4;
+  const int c_activeCol = 0;
+  const int c_nameCol = 1;
+  const int c_rangeCol = 2;
+  const int c_typeCol = 3;
```
Comment (cjh1) [inline tomviz/DataPropertiesModel.h]: Not sure about this prefix, is it used else where?

File: tomviz/DataSource.cxx [hunk 4]
```diff
@@ -612,6 +612,38 @@ QString DataSource::activeScalars() const
...
 }
 
+void DataSource::renameScalarsArray(const QString& oldName,
+                                    const QString& newName)
+{
+  const bool isCurrentScalars = oldName == activeScalars();
+
+  vtkAlgorithm* alg = algorithm();
+  if (alg == nullptr) {
+    return;
```
Comment (cjh1) [inline tomviz/DataSource.cxx]: Is this not an error ```alg``` is ```nullptr```?

File: tomviz/DataPropertiesPanel.cxx [hunk 5]
```diff
@@ -477,25 +498,24 @@ void DataPropertiesPanel::updateAxesGridLabels()
   }
 }
 
-void DataPropertiesPanel::updateActiveScalars()
+void DataPropertiesPanel::setActiveScalars(QString activeScalars)
```
Comment (cryos) [inline tomviz/DataPropertiesPanel.cxx]: Prefer const QString& when passing in strings

File: tomviz/DataPropertiesPanel.h [hunk 6]
```diff
@@ -52,7 +55,7 @@ private slots:
 
   void updateAxesGridLabels();
 
-  void updateActiveScalars();
+  void setActiveScalars(QString activeScalars);
```
Comment (cryos) [inline tomviz/DataPropertiesPanel.h]: As above

**Comment by cryos** [comment]
Just taking a look at this locally, there are a few clang-format issues picked up that it would be good to fix before we merge this.

File: tomviz/DataPropertiesPanel.h [hunk 7]
```diff
@@ -65,11 +68,15 @@ private slots:
...
   QPointer<pqProxyWidget> m_colorMapWidget;
   QPointer<QWidget> m_tiltAnglesSeparator;
+  DataPropertiesModel m_scalarsTableModel;
 
   void clear();
   void updateSpacing(int axis, double newLength);
-  void updateInformationWidget(QTreeWidget* infoTreeWidget,
-                               vtkPVDataInformation* dataInformation);
+  QList<ArrayInfo> getArraysInfo(vtkPVDataInformation* dataInfo) const;
+  void updateInformationWidget(QTableView* scalarsTable,
```
Comment (cryos) [inline tomviz/DataPropertiesPanel.h]: The QList parameters should be passed in as const references too, missed these first time around.

**Pull request merged**